AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -165,6 +193,15 @@ internal class CardViewController: FormViewController {
         }
         issuingCountryCode = binInfo.issuingCountryCode
         items.numberContainerItem.update(brands: brands)
+        
+        updateBillingAddressOptionalStatus(brands: brands)
+    }
+    
+    private func updateBillingAddressOptionalStatus(brands: [CardBrand]) {
+        let optionalBrands = configuration.billingAddress.optionalForBrands
+        let isOptional = optionalBrands.isDisjoint(with: brands.map(\.type)) == false
+        items.billingAddressItem.context.isOptional = isOptional
```

Comment:
what if `items.billingAddressItem` was not created (`billingAddressMode == .none`)?
I assume it will be created in memory, but not visible and then appear in `PaymentData` json as empty `billingaddress`?
Maybe we need IF or switch here?

Reply:
> it will never show up in the PaymentData

this is good, but `updateBillingAddressOptionalStatus` will trigger creation and/or apply changes to both `items.billingAddressItem` and `items.postalCodeItem`, even though they are not visible on a screen?